src/ziggurat/messaging/rabbitmq/cluster/producer.clj

```diff
@@ -0,0 +1,83 @@
+(ns ziggurat.messaging.rabbitmq.cluster.producer
+  (:require [clojure.tools.logging :as log]
+            [langohr.channel :as lch]
+            [langohr.basic :as lb]
+            [langohr.http :as lh]
+            [taoensso.nippy :as nippy]
+            [langohr.exchange :as le]
+            [ziggurat.messaging.rabbitmq.retry :refer :all]
+            [langohr.queue :as lq]
+            [clojure.string :as str]
+            [clojure.set :as set])
+  (:import (org.apache.kafka.common.header.internals RecordHeader)))
+
+(defn- record-headers->map [record-headers]
+  (reduce (fn [header-map record-header]
+            (assoc header-map (.key record-header) (String. (.value record-header))))
+          {}
+          record-headers))
+
+(defn- properties-for-publish
+  [expiration headers]
+  (let [props {:content-type "application/octet-stream"
+               :persistent   true
+               :headers      (record-headers->map headers)}]
+    (if expiration
+      (assoc props :expiration (str expiration))
+      props)))
+
+(defn publish
```

Comment:
why have you copied over the publish fn as well? Why not use the publish fn from the original implementation?